### DISCUSSION TITLE
Combine next-auth + next-intl middleware

### DISCUSSION THREAD

**Discussion by leaen** [vercel/next.js#51221] [Help]
### Summary

```
//next-intl middleware
import createMiddleware from 'next-intl/middleware';
 
export default createMiddleware({
  // A list of all locales that are supported
  locales: ['en', 'de'],
 
  // If this locale is matched, pathnames work without a prefix (e.g. `/about`)
  defaultLocale: 'en'
});
 
export const config = {
  // Skip all paths that should not be internationalized
  matcher: ['/((?!api|_next|.*\\..*).*)']
};

//next-auth middleware

export {default} from 'next-auth/middleware'

export const config = {
    matcher: ['/host/dashboard']
  }
```

I am trying to use both next-intl for internationalization and next-auth for authentication. Since in a middlware we can have a single config not able to figure out how to combine them.

I did found a solution here:
https://next-intl-docs.vercel.app/docs/next-13/middleware

But  having difficulty with that for accessing the protected pages its redirecting me to callback url which i dont want.

Moreover the matcher of next-auth middleware is very easy to use. Otherwise in every protected page I have to go with status variable way which I dont want to.

Any help appreciated. Thanks!

**Comment by Rohullah-Sarabi** [comment]
I had this problem. By use of this content I have solved it.
https://next-intl-docs.vercel.app/docs/routing/middleware#composing-other-middlewares

**Comment by Rohullah-Sarabi** [comment]
I had this problem. By use of this content I have solved it.
https://next-intl-docs.vercel.app/docs/routing/middleware#composing-other-middlewares
